Kernel/TTY/TTY.cpp

```diff
@@ -400,6 +409,34 @@ int TTY::set_termios(const termios& t)
         }
     }
 
+    if (m_termios.c_cflag & CS5)
+        attempt(change_character_size(FiveBits));
+    else if (m_termios.c_cflag & CS6)
+        attempt(change_character_size(SixBits));
+    else if (m_termios.c_cflag & CS7)
+        attempt(change_character_size(SevenBits));
+    else if (m_termios.c_cflag & CS8)
+        attempt(change_character_size(EightBits));
+
+    if (m_termios.c_cflag & CSTOPB)
+        attempt(change_stop_bits(Two));
+    else
+        attempt(change_stop_bits(One));
+
+    attempt(change_receiver_enabled(m_termios.c_cflag & CREAD));
+
+    if (!(m_termios.c_cflag & PARENB))
+        attempt(change_parity(None));
+    else if (m_termios.c_cflag & PARODD)
+        attempt(change_parity(Odd));
+    else
+        attempt(change_parity(Even));
+    // FIXME: add sticky parity stuff
+
+    // HUPCL is probably our job to handle
+
+    attempt(change_ignore_status_lines(m_termios.c_cflag & CLOCAL));
```

Comment:
shouldnt we just early return if any of the requested changes fail (instead of doing all of them regardless of one of them failing)